src/components/FileImporter.js

```diff
@@ -71,45 +71,41 @@ class FileImporter extends Nimiq.Observable {
     /**
      * @param {File} file
      */
-    _handleFile(file) {
+    async _handleFile(file) {
         this.$errorMessage.textContent = '';
+        this.$fileInput.value = '';
+
+        const url = URL.createObjectURL(file);
+        const image = this.$el.querySelector('img') || document.createElement('img');
+        image.classList.remove('pop-down'); // reset image style in case it was already used in UI
+        image.src = url;
+        const decoded = await this._readQrCode(image);
+
+        if (!decoded) {
+            AnimationUtils.animate('shake', this.$el);
+            this.$errorMessage.textContent = 'Could not read Login File.';
+            return;
+        }
+
+        if (this._displayFile) {
+            // Show image in UI
+            this.$el.appendChild(image);
+            image.classList.add('pop-down');
+        }
```

Comment:
Just a note about the new CSS classes behavior (this is not relevant in the way the importer is used currently, as `displayFile` is set to false, so the image is never displayed, but still):

When imorting a second image, while the first one is still displayed in the page (when `displayFile` is set to `true`), the second image is loaded into the page (`image.src = url`) first, and _then_ decoded by the qr-reader. Depending on how long this takes, the image is displayed - with full opacity and non-blurred - while the qr-reader is working. Only when the qr-reader returns, is the image animated with the pop-down animation (although it was already displayed in the animation's end state).

Before, the opacity and scale was at least set to the initial animation values while the qr-ready was working.

I fear that these changes will lead to UI-jumping from end state, to animation, to end state.

Reply:
This should not happen as the image is set to `display: none` if it doesn't have the `pop-down` class.

Reply:
Oh, you're right. I must have overlooked that. Disregard my comment :D